File: backend/src/Request/TravelFilterRequest.php
<?php

namespace App\Request;

class TravelFilterRequest
{
    private $type;

    private $launchCountry;

    private $destinationCountry;

    private $launchDate;

    private $arrivalDate;

    private $travelNumber;

    private $shipperID;

    private $carrierID;

    private $status;

    public function getType()
    {
        return $this->type;
    }

    public function getLaunchCountry()
    {
        return $this->launchCountry;
    }

    public function getDestinationCountry()
    {
        return $this->destinationCountry;
    }

    public function getLaunchDate()
    {
        return $this->launchDate;
    }

    public function getArrivalDate()
    {
        return $this->arrivalDate;
    }

    public function getTravelNumber()
    {
        return $this->travelNumber;
    }

    public function getShipperID()
    {
        return $this->shipperID;
    }

    public function getCarrierID()
    {
        return $this->carrierID;
    }

    public function getStatus()
    {
        return $this->status;
    }

}

File: flutter_boilerplate_dash/lib/module_travel/request/travel_filter_request.dart
class TravelFilterRequest {

  String type;
  String? launchCountry;
  String? destinationCountry;

  String status;

  String? launchDate;
  String? arrivalDate;

  String? travelNumber;

  TravelFilterRequest(
      {
        required this.type,
         this.launchCountry,
         this.destinationCountry,
         this.arrivalDate,
         this.launchDate,
       required  this.status,
        this.travelNumber
      });

  Map<String, dynamic> toJson() {
    return {
     'type': type,
     'launchCountry': launchCountry ?? '',
     'destinationCountry': destinationCountry ?? '',
     'launchDate': launchDate ?? '',
     'arrivalDate': arrivalDate ?? '',
     'status': status.isEmpty ? null :status,
      'travelNumber':travelNumber ??''
   };
  }
}